Core/Src/internalResistance.c

```diff
@@ -0,0 +1,221 @@
+        did not return any faulty data to the discrete buffer   */
+    if(currentDataGood)
+    {
+        // Average the discrete buffer and place the output in the average buffer
+        float currentDiscreteSum = 0;
+        for(int32_t i = 0; i < DISCRETE_CURRENT_BUFFER_SIZE; i++)
+        {
+            currentDiscreteSum += currentDiscreteBuffer[i];
+        }
+        currentAvgBuffer[avgBufferIndex] = currentDiscreteSum / DISCRETE_CURRENT_BUFFER_SIZE;
+
+        // Cycle through every BMB and every brick in the accumulator
+        for(int32_t i = 0; i < NUM_BMBS_PER_PACK; i++)
+        {
+            for(int32_t j = 0; j < NUM_BRICKS_PER_BMB; j++)
+            {
+                /*  An average can only be added to the buffer if the current sensor
+                    did not return any faulty data to the corresponding discrete buffer   */
+                if(voltageDataGood)
+                {
+                    // Average the discrete buffer and place the output in the average buffer
+                    float voltageDiscreteSum = 0;
+                    for(int32_t k = 0; k < DISCRETE_VOLTAGE_BUFFER_SIZE; k++)
+                    {
+                        voltageDiscreteSum += voltageDiscreteBuffer[i][j][k];
+                    }
+                    voltageAvgBuffer[i][j][avgBufferIndex] = voltageDiscreteSum / DISCRETE_VOLTAGE_BUFFER_SIZE;
+                }
+                else
+                {
+                    /*  If bad data was in the corresponding discrete voltage buffer, 
+                        it is filled with a placeholder value to indicate it cannot be used for calculation */
+                    voltageAvgBuffer[i][j][avgBufferIndex] = -1000.0f;
+
+                    // Static data good var is reset for the next discrete buffer filling cycle
+                    voltageDataGood[i][j] = true;
+                }
+            }    
+        }
+    }
+    else
+    {
+        /*  If bad data was in the corresponding discrete voltage buffer, 
+            it is filled with a placeholder value to indicate it cannot be used for calculation */
+        currentAvgBuffer[avgBufferIndex] = -1000.0f;
+
+        // Static data good var is reset for the next discrete buffer filling cycle
+        currentDataGood = true;
+    }
+
+    // Cycle the circular average buffers and loop if necessary
+    if (++avgBufferIndex >= AVERAGE_BUFFER_SIZE)
+    {
+        avgBufferIndex = 0;
+    }
+
+    // Max and min data initilized to far left and right values
+    float maxCurrent = -1000.0f;
+    float minCurrent = 1000.0f;
+
+    // Default max and min index is 0
+    uint32_t maxCurrentIndex = 0;
+    uint32_t minCurrentIndex = 0;
+
+    // Calculate max and min currents in buffer
+    for(int32_t i = 0; i < AVERAGE_BUFFER_SIZE; i++)
+    {
+        // If average current buffer containes placeholder value, -1000.0f, it is excluded from the max/min calc
+        if(currentAvgBuffer[i] > -900.0f)
```

Comment:
Why do a `>` check? If data is bad it should be exactly equal to `-1000.0`. If you replace this with a #define then this would just be `if (currentAvgBuffer[i] != BAD_DATA)`